### PR TITLE
feat: add InfiniteGrid methods

### PR DESCRIPTION
**Pull request by daybrush**
## Details

Added methods of InfiniteGrid.

Because the amount is large, I added the method first and tested it.

Events will be added and tested later.

### CODE REVIEW

File: src/GroupManager.ts [hunk 1]
```diff
@@ -53,6 +54,21 @@ export class GroupManager extends Grid<GroupManagerOptions> {
     }
   }
 
+  public getItemByKey(key: string | number): InfiniteGridItem | null {
+    return this.itemKeys[key] || null;
+  }
+  public getItemInfos(): InfiniteGridItemInfo[] {
+    return this.groupItems.map((item) => {
```
Comment (mixed) [inline src/GroupManager.ts:61]: How about it?
```js
return this.groupItems.map(({groupKey,key, element,html,data}) => {
      return {groupKey,key, element,html,data};
});
```

File: src/Infinite.ts [hunk 2]
```diff
@@ -59,15 +60,38 @@ export class Infinite extends Component<InfiniteEvents> {
...
+          startCursor: prevStartCursor,
+          endCursor: prevEndCursor,
+        });
+      } else {
+        this.trigger("requestAppend", {
+          groupKey: undefined,
+          startCursor: prevStartCursor,
+          endCursor: prevEndCursor,
+        });
+      }
```
Comment (mixed) [inline src/Infinite.ts:82]: How about it?
```js
        this.trigger(isDirectionEnd?"requestPrepend":"requestAppend", {
          groupKey: undefined,
          startCursor: prevStartCursor,
          endCursor: prevEndCursor,
        });
```

File: src/Infinite.ts [hunk 3]
```diff
@@ -152,10 +183,7 @@ export class Infinite extends Component<InfiniteEvents> {
     let nextEndCursor = -1;
 
     // sync cursors
-    [
-      ...result.changed,
-      ...result.maintained,
-    ].forEach(([prevIndex, nextIndex]) => {
+    result.maintained.forEach(([prevIndex, nextIndex]) => {
```
Comment (mixed) [inline src/Infinite.ts:186]: Is there always maintained?

**Review by mixed — APPROVED** [review]
LGTM

File: src/Infinite.ts (hunk 3 above)
Comment (daybrush) [reply src/Infinite.ts:186]: changed is maintained's subset.

File: src/InfiniteGrid.ts [hunk 4]
```diff
@@ -37,134 +126,323 @@ export class InfiniteGrid {
...
   }
-
-  public setCursors(startCursor: number, endCursor: number) {
+  /**
+   * Change the currently visible groups.
+   * @ko 현재 보이는 그룹들을 바꾼다.
+   * @param - first index of visible groups. <ko>보이는 그룹의 첫번째 index.</ko>
+   * @param - last index of visible groups. <ko>보이는 그룹의 마지막 index.</ko>
+   */
+  public setCursors(startCursor: number, endCursor: number): void {
```
Comment (WoodNeck) [inline src/InfiniteGrid.ts:229]: Looks like some methods are returning `this` while others returning `void`, is there any rules for this?

File: src/InfiniteGridItem.ts [hunk 5]
```diff
@@ -1,12 +1,28 @@
...
 export interface InfiniteGridItemStatus extends GridItemStatus {
   groupKey: string | number;
+  key: string | number;
+  html: string;
 }
-export class InfiniteGridItem extends GridItem {
+
+export class InfiniteGridItem extends GridItem implements Required<InfiniteGridItemInfo> {
   public groupKey: string | number;
+  public html: string;
```
Comment (WoodNeck) [inline src/InfiniteGridItem.ts:12]: How about adding `readonly` to these props if they're immutable?

**Review by WoodNeck — APPROVED** [review]

File: src/Infinite.ts [hunk 6]
```diff
@@ -59,15 +60,38 @@ export class Infinite extends Component<InfiniteEvents> {
...
+        this.trigger("requestAppend", {
+          groupKey: undefined,
+          startCursor: prevStartCursor,
+          endCursor: prevEndCursor,
+        });
+      }
+      return;
+    } else if (prevStartCursor === -1 || prevEndCursor === -1) {
+      const nextCursor = isDirectionEnd ? 0 : length - 1;
+      this.trigger("change", {
```
[the remark this answers is not in the document]
Comment (daybrush) [reply src/Infinite.ts:86]: @WoodNeck 

Events of InfiniteGrid will be added, but events of internal modules are unlikely to be added separately.

**Pull request merged**